src/collection.ts

```diff
@@ -764,10 +771,23 @@ export class Collection<TSchema extends Document = Document> {
     filter: Filter<TSchema> = {},
     options: CountDocumentsOptions = {}
   ): Promise<number> {
-    return await executeOperation(
-      this.client,
-      new CountDocumentsOperation(this as TODO_NODE_3286, filter, resolveOptions(this, options))
-    );
+    const pipeline = [];
+    pipeline.push({ $match: filter });
+
+    if (typeof options.skip === 'number') {
+      pipeline.push({ $skip: options.skip });
+    }
+
+    if (typeof options.limit === 'number') {
+      pipeline.push({ $limit: options.limit });
+    }
+
+    pipeline.push({ $group: { _id: 1, n: { $sum: 1 } } });
+
+    const cursor = this.aggregate<{ n: number }>(pipeline, options);
+    const doc = await cursor.next();
+    await cursor.close();
+    return doc?.n ?? 0;
```

Comment:
```suggestion
    try {
      const doc = await cursor.next();
      return doc?.n ?? 0;
    } finally {
       await cursor.close().catch(error => squashError(error);
    }
```

Do you think we need to be careful about closing the cursor?  I'm thinking it can't hurt 🤷

Reply:
I followed how findOne was written, I recall there being a reason but I can't recall and I can't say I disagree so sure 🤔

Reply:
I recall now, the expectation is that our cursor's next cleans itself up if it throws and we also did not want to squash errors from close because we expect it not to throw. The thinking is both of these expectations are part of the API, if they aren't satisfied then something is broken and if we squash / use finally we are covering up a problem that needs fixing. 

Added testing!